### PR TITLE
RHCLOUD-22545 Split the Camel and RHOSE processors

### PR DESCRIPTION
**Pull request by gwenneg**
This PR:
- introduces a clean separation of the Camel and RHOSE processors
- refactors everything to make the code simpler and easier to test/maintain
- does not contain any major tests changes to ensure that the refactoring didn't break something

`CamelTypeProcessorTest` will be refactored as well, later.

### CODE REVIEW

**Comment by gwenneg** [comment]
Good luck with the review 😁

**Comment by josejulio** [comment]
> Good luck with the review grin

image

File: engine/src/main/java/com/redhat/cloud/notifications/processors/rhose/RhoseTypeProcessor.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        outgoingEvent.put(ORG_ID_FILTER_NAME, endpoint.getOrgId());
+        outgoingEvent.put(PROCESSOR_NAME, extras.get(PROCESSOR_NAME));
+        outgoingEvent.put("originaleventid", originalEventId);
+        outgoingEvent.put("data", data);
+
+        return outgoingEvent;
+    }
+
+    private void sendEvent(JsonObject outgoingEvent) {
+        // TODO Can we cache and reuse this instance instead of building a new one on each call?
```
Comment (josejulio) [inline engine/src/main/java/com/redhat/cloud/notifications/processors/rhose/RhoseTypeProcessor.java:167]: Sounds like a good idea - something we could try

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**